server/svix-server/src/error.rs

```diff
@@ -455,3 +427,22 @@ impl From<crate::core::webhook_http_client::Error> for Error {
         }
     }
 }
+
+/// Utility function for Converting a [`DbErr`] into an [`HttpError`] (wrapped by [`Error`]) on the
+/// error "duplicate key value violates unique constraint". This is to be used in `map_err` calls
+/// on creation/update of records
+pub fn http_error_on_conflict(db_err: DbErr) -> Error {
+    match db_err {
+        DbErr::Query(RuntimeErr::SqlxError(e))
+            if e.as_database_error()
+                .and_then(|e| e.code())
+                .filter(|code| code == "23505")
+                .is_some() =>
+        {
+            HttpError::conflict(None, None).into()
+        }
+        // This always inserts this line's locaton, but because it should be wrapped by a [`ctx`]
+        // call, you should still get a meaningful trace
+        e => Error::database(e, location!()),
```

Comment:
Maybe it's better use a `""` instead of `location!`? I'm just thinking from a debugging perspective, for something like

```rust
ctx!(app.insert(txn).await.map_err(http_error_on_conflict))?
```

The line # in `error.rs` might just be noise.